utoipa-gen/src/lib.rs

```diff
@@ -1081,11 +1142,38 @@ impl Parse for Type<'_> {
             parse_array(input)
         }?;
 
-        Ok(Type {
+        Ok(Self(Type {
             ty: Cow::Owned(ty),
             is_array,
             is_option,
-        })
+            is_inline: false,
+        }))
+    }
+}
+
+impl Parse for Type<'_> {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        const EXPECTED_TYPE_DEFINITION: &str =
+            "unexpected attribute, expected `inline(Type)` or `Type`, where `Type` can be `Type`, `[Type]` or `Option<Type>`";
+
+        // Try parsing as `inline(Type)`
+        if input.fork().parse::<InlineType>().is_ok() {
+            let t: Self = input.parse::<InlineType>()?.0;
+            return Ok(t);
+        }
+
+        // Try parsing as `Type`, `[Type]` or `Option<Type>`)
+        let t: Type = input
+            .parse::<ArrayOrOptionType>()
+            .map_err(|error| {
+                syn::Error::new(
+                    Span::call_site(),
```

Comment:
Returned error will be a syn error which contains span, it focuses the error better for user. :)
```suggestion
                    error.span(),
```